### PR TITLE
[Jupyter] Data persistent

### PR DESCRIPTION
**Pull request by eliyahu77**
add support in mlrunce  jupyter data persistent

### CODE REVIEW

File: dockerfiles/jupyter/mlcestart.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    echo "base data does not exist."
+    echo "1" > $file_path
+    echo "base data exists."
+    cd / ; tar -xvf /tmp/basehome.tar
+    # Perform actions when the file does not exist
+fi
+
+cd ${HOME}
+
+start-notebook.sh  --ip=0.0.0.0 --port=8888 --NotebookApp.token="" --NotebookApp.password="" --NotebookApp.default_url=/lab
```
Comment (alonmr) [inline dockerfiles/jupyter/mlcestart.sh:16]: what about the mlrun db? you need to run them in parallel like in the dockerfile

File: dockerfiles/jupyter/Dockerfile [hunk 2]
```diff
@@ -25,6 +25,15 @@ RUN apt-get update && \
 RUN apt-get update && apt-get install --no-install-recommends -y \
   graphviz \
  && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update && apt-get install -y curl apt-transport-https
```
Comment (alonmr) [inline dockerfiles/jupyter/Dockerfile:29]: why is this needed?

File: dockerfiles/jupyter/Dockerfile [hunk 3]
```diff
@@ -25,6 +25,15 @@ RUN apt-get update && \
...
+# Download and install kubectl
+RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && \
```
Comment (alonmr) [inline dockerfiles/jupyter/Dockerfile:31]: why is this needed?

File: dockerfiles/jupyter/mlcestart.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+file_path="${HOME}/.intdata"
+
+if [ -f "$file_path" ]; then
```
Comment (alonmr) [inline dockerfiles/jupyter/mlcestart.sh:3]: please explain what is the base data? when will it exists and when will it not?

**Review by alonmr — COMMENTED** [review]
A few comments and questions

File: dockerfiles/jupyter/mlcestart.sh (hunk 1 above)
Comment (eliyahu77) [reply dockerfiles/jupyter/mlcestart.sh]: not sure i understand the comment

File: dockerfiles/jupyter/Dockerfile (hunk 2 above)
Comment (eliyahu77) [reply dockerfiles/jupyter/Dockerfile:29]: need fot install deps such as kubectl

File: dockerfiles/jupyter/Dockerfile (hunk 3 above)
Comment (eliyahu77) [reply dockerfiles/jupyter/Dockerfile:31]: this is kubectl in order to be able to  interact with k8s cluster

File: dockerfiles/jupyter/Dockerfile [hunk 5]
```diff
@@ -25,6 +25,15 @@ RUN apt-get update && \
...
+    chmod +x kubectl && \
+    mv kubectl /usr/local/bin/
+
+
+
```
Comment (alonmr) [inline dockerfiles/jupyter/Dockerfile:36]: Too many spaces
```suggestion
```

File: dockerfiles/jupyter/Dockerfile [hunk 6]
```diff
@@ -69,6 +78,10 @@ ENV JUPYTER_ENABLE_LAB=yes \
     MLRUN_HTTPDB__LOGS_PATH=$HOME/data/logs \
     MLRUN_ENV_FILE=$HOME/mlrun.env
 
+RUN tar -cvf /tmp/basehome.tar $HOME
```
Comment (alonmr) [inline dockerfiles/jupyter/Dockerfile:81]:
```suggestion
# backup home since it will be deleted when using pvc
RUN tar -cvf /tmp/basehome.tar $HOME
```

File: dockerfiles/jupyter/mlcestart.sh [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    echo "base data exists."
+    # Perform actions when the file exists
```
Comment (alonmr) [inline dockerfiles/jupyter/mlcestart.sh:5]:
```suggestion
file_path="${HOME}/.intdata"

# .intdata indicates if we extracted the home tar or not
if [ -f "$file_path" ]; then
    echo "Base data exists."
```

File: dockerfiles/jupyter/mlcestart.sh [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+else
+    echo "base data does not exist."
+    echo "1" > $file_path
+    echo "base data exists."
+    cd / ; tar -xvf /tmp/basehome.tar
+    # Perform actions when the file does not exist
```
Comment (alonmr) [inline dockerfiles/jupyter/mlcestart.sh:11]:
```suggestion
    # Perform actions when the file does not exist
    echo "Base data does not exist, extracting home backup..."
    cd / ; tar -xvf /tmp/basehome.tar
    echo "1" > $file_path
```

**Review by alonmr — APPROVED** [review]
